onnxruntime/core/graph/contrib_ops/contrib_defs.cc

```diff
@@ -1923,6 +1923,248 @@ Matrix product that behaves like numpy.matmul: https://docs.scipy.org/doc/numpy-
+            auto& dim1 = dims1.Get(i);
+            if (dim0 != dim1) {
+              fail_shape_inference("Input dims at: ", i, " do not match");
+            }
+          }
+          propagateShapeFromInputToOutput(ctx, 0, 0);
+        });
+
+    static const char* Binarizer_doc = R"DOC(
+The operator will accept a single sparse tensor input and will output
+a sparse tensor of the same shape and type. For each of the elements it
+would compare the value of the 'threshold' attribute and for all values above the threshold
+it would set the output value for the same index to T(1) and T(0) otherwise.
+
+XXX: Can output be simply an integer or a boolean?
+)DOC";
+
+  ONNX_CONTRIB_OPERATOR_SCHEMA(Binarizer)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "A", "A sparse matrix of numeric type", "T")
+      .Attr(
+        "threshold",
+        "A boundary value",
+        AttributeProto::INT,
+        static_cast<int64_t>(0))
+       .Output(0, "Y", "A sparse tensor with 1 or 0 for each of the input values", "T")
+       .TypeConstraint("T", {"sparse_tensor(float)"}, "Constrain input and output float at this time")
+       .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+          propagateElemTypeFromInputToOutput(ctx, 0, 0);
+          propagateShapeFromInputToOutput(ctx, 0, 0);
+        });
+
+    static const char* Bucketizer_doc = R"DOC(
+This operator behaves similar to https://pytorch.org/docs/stable/generated/torch.bucketize.html
+XXX: the output would be a sparse tensor of the same shape, but each value in it would be replaced
+to an index of the corresponding bucket.
+XXX: The output type attribute will be omitted for now. Fixed to int64_t.
+)DOC";
+  ONNX_CONTRIB_OPERATOR_SCHEMA(Bucketizer)
+      .SetDomain(kMSDomain)
+      .SinceVersion(1)
+      .Input(0, "A", "A sparse matrix of numeric type", "T")
+      .Input(0, "B", "1-D Boundaries tensor", "T1")
+        .Attr(
+            "tolerance",
+            "A boundary value",
+            AttributeProto::FLOAT,
+            std::numeric_limits<float>::epsilon())
+       .Output(0, "Y", "Sparse tensor where values are indices to the boundaries tensor", "T")
+      .TypeConstraint("T", {"sparse_tensor(float)"}, "Constrain input to float at this time")
+      .TypeConstraint("T1", {"sparse_tensor(int64)"}, "index type")
+        .TypeAndShapeInferenceFunction([](ONNX_NAMESPACE::InferenceContext& ctx) {
+          // Output type is index type
+          propagateElemTypeFromDtypeToOutput(ctx, TensorProto_DataType_INT64, 0);
+          propagateShapeFromInputToOutput(ctx, 0, 0);
+        });
+
+    static const char* OneHotEncoder_doc = R"DOC(
+The operator implements a simplified version of sklearn OneHotEncoder
+https://scikit-learn.org/stable/modules/generated/sklearn.preprocessing.OneHotEncoder.html
+`categories` attribute contains an array of category values.
+
+The input is a sparse tensor of values.
```

Comment:
I responded to your comment on the word doc. Pasting the example sklearn code usage that'd match our expectation:

```
>>> seq = np.linspace(0, 2002, num=2003, dtype=int) 

>>> enc = OneHotEncoder(categories=[seq], handle_unknown='ignore') 

>>> enc.fit([[1]]) # arbitrary, to ensure categories are populated as inputted earlier 

OneHotEncoder(categories=[array([   0,    1,    2, ..., 2000, 2001, 2002])], 

              handle_unknown='ignore') 

>>> encoded_matrix_as_list = enc.transform([[1001], [2002]]).todense().tolist() 

>>> encoded_matrix_as_list[0][1001] 

1.0 

>>> encoded_matrix_as_list[1][2002] 

1.0 
```